api/script/index.js

```diff
@@ -9,31 +9,55 @@ var currency = require('../../currency');
 var processReversal = (bus, log) => params => {
     var transferId;
 
-    var portReversal = (port, reversal) => bus.importMethod(`${port}.${reversal.transferType}.${reversal.operation}`)(reversal);
+    var portReversal = (port, reversal) => {
+        if (port && reversal.transferType && reversal.operation) {
+            return bus.importMethod(`${port}.${reversal.transferType}.${reversal.operation}`)(reversal);
+        } else {
+            return Promise.resolve(reversal);
+        }
+    };
+
+    var dbPushReverse = reverse => bus.importMethod('db/transfer.push.reverse')(reverse)
+        .then(pushResult => {
+            return reverse;
+        });
 
     var reverse = reversal => {
-        reversal = reversal && reversal[0] && reversal[0][0];
-        if (reversal) {
-            transferId = reversal.transferId;
+        transferId = reversal.transferId;
+        if (reversal && !reversal.reversed) { // reverse only transaction that have NOT been reversed
             reversal.udfAcquirer && (reversal.udfAcquirer.mti = reversal.mti);
             reversal.amount = {
                 transfer: currency.amount(reversal.transferCurrency, reversal.transferAmount)
             };
-        }
-        return reversal && portReversal(reversal.issuerPort, reversal)
-        .then(result => {
-            if (reversal.issuerPort === reversal.ledgerPort) {
-                return result;
-            } else {
-                return portReversal(reversal.ledgerPort, reversal)
-                .then(() => result);
+
+            // prepare reversal object for postReversal
+            if (!reversal.operation) {
+                reversal.operation = 'reverse';
             }
-        });
+            if (!reversal.transferType) {
+                reversal.transferType = 'push';
+            }
+
+            return reversal && portReversal(reversal.issuerPort, reversal)
+            .then(result => {
+                if (reversal.issuerPort === reversal.ledgerPort) {
+                    return reversal;
+                } else {
+                    return portReversal(reversal.ledgerPort, reversal)
+                    .then(() => reversal);
+                }
+            });
+        } else {
+            throw errors.notFound();
```

Comment:
When transaction was found and was already reversed, we need to throw different error.